jackson/src/main/scala/com/lightbend/lagom/internal/jackson/JacksonJsonSerializer.scala

```diff
@@ -87,14 +86,16 @@ private[lagom] class JacksonJsonSerializer(val system: ExtendedActorSystem)
     val startTime = if (isDebugEnabled) System.nanoTime else 0L
     val compressed = isGZipped(bytes)
 
-    val migration = migrations.get(manifest)
+    val (fromVersion, manifestClassName) = parseManifest(manifest)
+
+    val migration = migrations.get(manifestClassName)
```

Comment:
This is introducing a regression (that most probably was also introduced in the Scala variation).

We can not run a specific migration for a given version anymore. 
For instance:

```
lagom.serialization.json.migrations { 
  "com.lightbend.lagom.serialization.Event1" = "com.lightbend.lagom.serialization.TestEventMigration"
  "com.lightbend.lagom.serialization.Event1#3" = "com.lightbend.lagom.serialization.TestEventMigration2"
}
```
Note that `Event1#3` is explicitly using `TestEventMigration2`.

With this PR, `TestEventMigration` will always be selected for ` Event1#3`.

This can be fixed by replacing line 91 by:
```scala
val migration = migrations.get(manifest).orElse(migrations.get(manifestClassName))
```

If we don't find a explicit migration for `manifest` we fall back to `manifestClassName`